File: leetcode/0350. Intersection of Two Arrays II/Solution.swift
// https://leetcode.com/problems/intersection-of-two-arrays-ii/

class Solution {
    func intersect(_ nums1: [Int], _ nums2: [Int]) -> [Int] {
        let nums1 = nums1.sorted()
        let nums2 = nums2.sorted()

        var i = 0, j = 0
        var r = [Int]()
        while i < nums1.count && j < nums2.count {
            if nums1[i] < nums2[j] {
                i += 1
            } else if nums1[i] > nums2[j] {
                j += 1
            } else {
                r.append(nums1[i])
                i += 1
                j += 1
            }
        }

        return r
    }
}

let s = Solution()
assert(s.intersect([1,2,2,1], [2,2]).sorted() == [2,2])
assert(s.intersect([4,9,5], [9,4,9,8,4]).sorted() == [4,9])

File: leetcode/0350. Intersection of Two Arrays II/Solution.py
# https://leetcode.com/problems/intersection-of-two-arrays-ii/

from typing import List

class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1 = sorted(nums1)
        nums2 = sorted(nums2)

        i, j = 0, 0
        r = []
        while i < len(nums1) and j < len(nums2):
            if nums1[i] < nums2[j]:
                i += 1
            elif nums1[i] > nums2[j]:
                j += 1
            else:
                r.append(nums1[i])
                i += 1
                j += 1

        return r

s = Solution()
assert sorted(s.intersect([1,2,2,1],[2,2])) == [2,2]
assert sorted(s.intersect([4,9,5],[9,4,9,8,4])) == [4,9]
